Library/Formula/agda.rb

```diff
@@ -20,14 +18,17 @@ class Agda < Formula
   head "https://github.com/agda/agda.git", :branch => "master"
 
   option "without-stdlib", "Don't install the Agda standard library"
-  option "with-malonzo-ffi",
-    "Include the MAlonzo backend's FFI (depends on the standard library)"
+  option "without-malonzo", "Don't install GHC (needed for the MAlonzo backend)"
 
-  depends_on "ghc" => :build
+  if build.with? "malonzo"
+    depends_on "ghc"
+  else
+    depends_on "ghc" => :build
+  end
   depends_on "cabal-install" => :build
 
   depends_on "gmp"
-  depends_on "emacs" => :optional
+  depends_on "emacs" => :recommended
```

Comment:
Why? The default was wrong, and cause the problem with the Emacs mode to go unnoticed for a long time, since it wasn’t exercised by CI.

The Agda developers assume [the user is working in Emacs](http://wiki.portal.chalmers.se/agda/pmwiki.php?n=Main.QuickGuideToEditingTypeCheckingAndCompilingAgdaCode):

> Agda programs are commonly edited using Emacs. To edit a module (assuming you have installed Agda and its Emacs mode properly), start Emacs and open a file ending in .agda. Programs are developed interactively, which means that one can type check code which is not yet complete: if a question mark (?) is used as a placeholder for an expression, and the buffer is then checked, Agda will replace the question mark with a “hole” which can be filled in later.